src/main/java/com/aws/greengrass/localdebugconsole/DashboardServer.java

```diff
@@ -47,18 +51,34 @@ public class DashboardServer extends WebSocketServer implements KernelMessagePus
             new DefaultConcurrentHashMap<>(HashSet::new);
     private final DefaultConcurrentHashMap<String, Set<WebSocket>> logWatchlist =
             new DefaultConcurrentHashMap<>(HashSet::new);
+    private final DefaultConcurrentHashMap<String, Set<WebSocket>> pubSubWatchList =
+            new DefaultConcurrentHashMap<>(HashSet::new);
     @Getter(AccessLevel.PACKAGE)
     private final CompletableFuture<Object> started = new CompletableFuture<>();
     private final Authenticator authenticator;
 
+    PubSubIPCEventStreamAgent pubSubIPCAgent;
+
+    private final Consumer<PublishEvent> pubSubCallback = (message) -> {
+        pubSubWatchList.computeIfPresent(message.getTopic(), (k, set) -> {
+            for (WebSocket conn : set) {
+                CommunicationMessage resMessage = new CommunicationMessage(message.getTopic(), new String(message.getPayload()));
+                sendIfOpen(conn, new Message(MessageType.PUB_SUB_MSG, -1, resMessage));
```

Comment:
Nit: You need not pass request id (-1) here.

Reply:
why not?